### PR TITLE
Feat(cta): render lightbox video player to ctas

### PR DESCRIPTION
**Pull request by annawen1**
### Related Ticket(s)

Create the link version of the video component that opens into the Lightbox media viewer #1111

### Description

add `video` type to the `CTA` and render lightbox when clicked.

Mar-30-2020 14-01-22

### Changelog

**Changed**

- call videoplayer api in `CTA` component 
- add `PlayOutline20` icon to be used for `video` types

### CODE REVIEW

File: packages/react/src/components/CTA/CTA.js [hunk 1]
```diff
@@ -26,11 +33,46 @@ const { prefix } = settings;
...
+const CTA = ({ style, type, customClassName, ...otherProps }) => {
+  const [renderLightBox, openLightBox] = useState(false);
+  const [videoTitle, setVideoTitle] = useState([{ title: '', key: 0 }]);
+  const [mediaData, setMediaData] = useState({});
+
+  useEffect(() => {
+    video();
+  }, [video, style, type]);
+
+  const video = useCallback(async () => {
```
Comment (jeffchew) [inline packages/react/src/components/CTA/CTA.js]: The const `video` seems pretty generic, this method seems to be making the call to get the meta data then setting the title. Perhaps rename to `getVideoData` or something along those lines?

File: packages/react/src/components/CTA/CTA.js [hunk 2]
```diff
@@ -136,6 +347,7 @@ const _renderButtons = ({ buttons }) =>
  * @returns {*} object
  */
 const _renderFeatureCard = featureCard => {
+  if (featureCard.type === 'video') featureCard.cta.href = '#';
```
Comment (jeffchew) [inline packages/react/src/components/CTA/CTA.js]: Would this run into issues if there were actual anchor links on the page? Like if the link was inside of the Table of Contents?

**Comment by annawen1** [comment]
> @annawen1 had an overall question regarding the anchor link being set as `#`, I think that might pose issues on pages with anchor links and can take users to the top of the page unintentionally?
> 
> Also, because CTA is starting to look a little loaded, I'm wondering if it's worth thinking about breaking down CTA and have sub components with corresponding type handlers in their own subcomponent files within CTA?

@jeffchew yeah I was setting the hrefs to `#` because if I don't pass anything, the links won't render clickable. 

And yep I was talking to @kennylam earlier and I think we should probably go back in to refactor since it seems like it's only going to get more complicated

File: packages/react/src/components/CTA/CTA.js (hunk 1 above)
Comment (annawen1) [reply packages/react/src/components/CTA/CTA.js]: yep gotcha

**Comment by jeffchew** [comment]
> @jeffchew yeah I was setting the hrefs to `#` because if I don't pass anything, the links won't render clickable.

Is there another solution we could use?
 
> And yep I was talking to @kennylam earlier and I think we should probably go back in to refactor since it seems like it's only going to get more complicated

How much time would that take?

**Comment by jeffchew** [comment]
> > And yep I was talking to @kennylam earlier and I think we should probably go back in to refactor since it seems like it's only going to get more complicated
> 
> How much time would that take?
> 
> @jeffchew I can open another branch based off of this one - I'm thinking a day or two to separate things up?

Wish we could do threads on the main comments. I'm fine with having it in a separate PR. The other question about the anchor links, I doublechecked and it looks like the `preventDefault`'s you have should prevent the anchor jump behavior from what I can tell...

File: packages/react/src/components/CTA/CTA.js (hunk 2 above)
Comment (jeffchew) [reply packages/react/src/components/CTA/CTA.js]: Resolving, looks like `preventDefault` will take care of this.

**Comment by photodow** [comment]
> Also, because CTA is starting to look a little loaded, I'm wondering if it's worth thinking about breaking down CTA and have sub components with corresponding type handlers in their own subcomponent files within CTA?

I can't think of any issues off hand with having sub-components. I think the important thing to maintain are the types and styles, so we don't have to repeat that logic everywhere.

File: packages/react/src/components/CTA/CTA.js [hunk 3]
```diff
@@ -26,11 +33,46 @@ const { prefix } = settings;
...
+const CTA = ({ style, type, customClassName, ...otherProps }) => {
+  const [renderLightBox, openLightBox] = useState(false);
+  const [videoTitle, setVideoTitle] = useState([{ title: '', key: 0 }]);
+  const [mediaData, setMediaData] = useState({});
+
+  useEffect(() => {
+    getVideoData();
+  }, [getVideoData, style, type]);
+
+  const getVideoData = useCallback(async () => {
```
Comment (jeffchew) [inline packages/react/src/components/CTA/CTA.js]: Minor thing, this could use JSDocs.

Comment (annawen1) [reply packages/react/src/components/CTA/CTA.js]: coming right up

**Pull request merged**